neevaco/neeva-ios
# Add parent tab to `createOrSwitchToTab`

We want to connect back button to cheatsheet, which opens url with `createOrSwitchToTab`.

Changes:
- Add parent tab argument to `createOrSwitchToTab`
- Add parent tab argument to `getTabFor(url:)`
  - if url matches, then verify parent tab if given a parent tab
  - use `parentUUID` for zombie tabs
- Pass on parent tab argument when calling `addTab`
- Use currently selected tab in cheatsheet as parent tab

PS:
1) it seems like tab group comes for free just by setting parent property
2) if tab A opens url B and url C, and then tab with url B also opens url C, there will be two tabs with urls because they will have different parents.

## Checklists

### How was this tested?

### Manual test cases
1) Open url in cheatsheet. Verify back button is enabled. Verify that back button behaves with parent tab
2) Open url in cheatsheet. Go back to the first page, open the same url in cheatsheet. Verify that only one tab is created

### Screenshots and screen recordings
| Case 1 | Case 2 |
| ------ | ------ |
| <video src="https://user-images.githubusercontent.com/44557675/176306968-4888ad50-d25c-4120-a041-22593a32072f.MOV"> | <video src="https://user-images.githubusercontent.com/44557675/176306985-de75bb53-1575-4b21-a689-f5e8aa8f39b2.MOV"> |
## Issues addressed
